### PR TITLE
Eventsource trace sinks

### PR DESCRIPTION
**Pull request by djfr**
This PR is way bigger then it should be, because I've been going at this for at least 3 weeks, crazy interruptions ...

This adds a static ETW and Trace sinks to BigBrother, through a static Write method and adds the ability to sink normal BigBrother exception events to both ETW and Trace.

### CODE REVIEW

File: src/Tests/DevOpsFlex.Telemetry.Tests/SingleReplayCastTest.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        replay.Dispose();
+
+        connection.IsDisposed.Should().BeTrue();
+        subscription.IsDisposed.Should().BeTrue();
+
+        replay.Replay.IsDisposed.Should().BeTrue();
+    }
+}
+
+public class Disposable : IDisposable
```
Comment (RuaidhriMcDonnell) [inline src/Tests/DevOpsFlex.Telemetry.Tests/SingleReplayCastTest.cs]: Why a class? Why not a mock & a verify the method was called?

Comment (djfr) [reply src/Tests/DevOpsFlex.Telemetry.Tests/SingleReplayCastTest.cs]: Fair point, fixed it to use a normal Mock<IDisposable> instead!

File: src/DevOpsFlex.Telemetry/SingleReplayCast.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        /// <summary>
+        /// Subscribes to a single replay buffer.
+        ///     This will check if there's already a single active subscription, and if it is returns the existing one instead.
+        /// </summary>
+        /// <param name="action">The <see cref="Action{T}"/> we want to perform during the subscription.</param>
+        /// <returns>The subscription <see cref="IDisposable"/>.</returns>
+        public IDisposable Subscribe(Action<T> action)
+        {
+            lock (Gate)
```
Comment (RuaidhriMcDonnell) [inline src/DevOpsFlex.Telemetry/SingleReplayCast.cs]: could implement a double checked locking pattern for performance

Comment (djfr) [reply src/DevOpsFlex.Telemetry/SingleReplayCast.cs]: This is called only on a double subscription, which shouldn't happen when called correctly, so it's more of a defence gate. I'll leave it as a single lock.

**Pull request merged**